src/NServiceBus.Core.Tests/Recoverability/DefaultRecoverabilityPolicyTests.cs

```diff
@@ -82,21 +106,76 @@ public void When_immediate_retries_turned_off_and_delayed_retry_not_available_sh
         public void When_slr_counter_header_exists_recoverability_policy_should_use_it()
         {
             var policy = CreatePolicy(maxImmediateRetries: 0, maxDelayedRetries: 1, delayedRetryDelay: TimeSpan.Zero);
-            var errorContext = CreateErrorContext(headers: new Dictionary<string, string> { { Headers.Retries, "1" } });
+            var errorContext = CreateErrorContext(retryNumber: 1);
 
-            var recoverabilityAction = policy.Invoke(errorContext);
+            var recoverabilityAction = policy(errorContext);
 
             Assert.IsInstanceOf<MoveToError>(recoverabilityAction, "When slr cunter in headers reaches max slr retries, policy should return MoveToErrors");
         }
 
-        ErrorContext CreateErrorContext(int numberOfDeliveryAttempts = 0, Dictionary<string, string> headers = null)
+        [Test]
+        public void ShouldRetryTheSpecifiedTimesWithIncreasedDelay()
+        {
+            var baseDelay = TimeSpan.FromSeconds(10);
+            var policy = CreatePolicy(maxImmediateRetries: 0, maxDelayedRetries: 2, delayedRetryDelay: baseDelay);
+
+            var errorContext = CreateErrorContext(retryNumber: 0);
+            var result1 = (DelayedRetry) policy(errorContext);
+
+            errorContext = CreateErrorContext(retryNumber: 1);
+            var result2 = (DelayedRetry) policy(errorContext);
+
+            errorContext = CreateErrorContext(retryNumber: 2);
+            var result3 = policy(errorContext);
+
+
+            Assert.AreEqual(baseDelay, result1.Delay);
+            Assert.AreEqual(TimeSpan.FromSeconds(20), result2.Delay);
+            Assert.IsInstanceOf<MoveToError>(result3);
+        }
+
+        [Test]
+        public void ShouldCapTheRetryMaxTimeTo24Hours()
+        {
+            var provider = DefaultRecoverabilityPolicy.CurrentUtcTimeProvider;
+            try
+            {
+                var now = DateTime.UtcNow;
+                var baseDelay = TimeSpan.FromSeconds(10);
+                DefaultRecoverabilityPolicy.CurrentUtcTimeProvider = () => now;
+
+                var policy = CreatePolicy(maxImmediateRetries: 0, maxDelayedRetries: 2, delayedRetryDelay: baseDelay);
+
+                var moreThanADayAgo = now.AddHours(-24).AddTicks(-1);
+                var headers = new Dictionary<string, string>
+                {
+                    {Headers.RetriesTimestamp, DateTimeExtensions.ToWireFormattedString(moreThanADayAgo)}
+                };
+
+                var errorContext = CreateErrorContext(headers: headers);
+
+                var result = policy(errorContext);
+
+                Assert.IsInstanceOf<MoveToError>(result);
+            }
+            finally
+            {
+                DefaultRecoverabilityPolicy.CurrentUtcTimeProvider = provider;
```

Comment:
Can we try to remove `CurrentUtcTimeProvider`. Currently it's used only in this place so we actually do not need it. If we do in the future I would propose to have Invoke override in the `DefaultRecoverabilityPolicy` that will take `now` as additional parameter